program/rust/src/time_machine_types.rs

```diff
@@ -6,20 +6,375 @@ use crate::c_oracle_header::{
+                    inflated_prev_conf,
+                    inflated_current_price,
+                    inflated_current_conf,
+                    update_time,
+                    prev_entry,
+                )?;
+            }
+            1 => {
+                //initialize next entry to be the same as the current one
+
+                self.running_price[current_entry] = self.running_price[prev_entry];
+                self.running_confidence[current_entry] = self.running_confidence[prev_entry];
+
+                //update price
+                let time_to_entry_end = Self::get_time_to_entry_end(prev_time);
+                let time_after_entry_end = update_time - time_to_entry_end;
+                let entry_end_price = Self::weighted_average(
+                    time_after_entry_end,
+                    inflated_prev_price,
+                    time_to_entry_end,
+                    inflated_current_price,
+                )?;
+                let entry_end_conf = Self::weighted_average::<u64>(
+                    try_convert::<i64, u64>(time_after_entry_end)?,
+                    inflated_prev_conf,
+                    try_convert::<i64, u64>(time_to_entry_end)?,
+                    inflated_current_conf,
+                )?;
+
+                self.add_price_to_entry(
+                    inflated_prev_price,
+                    inflated_prev_conf,
+                    entry_end_price,
+                    entry_end_conf,
+                    time_to_entry_end,
+                    prev_entry,
+                )?;
+
+                //update current entry validity
+                self.entry_validity[prev_entry] = if self.max_update_time_gap >= THRESHOLD {
+                    0
+                } else {
+                    self.valid_entry_counter += 1;
+                    1
+                };
+
+                //update max_update_time
+                self.max_update_time_gap = update_time;
+
+
+                self.add_price_to_entry(
+                    inflated_prev_price,
+                    inflated_prev_conf,
+                    inflated_current_price,
+                    inflated_current_conf,
+                    update_time,
+                    current_entry,
+                )?;
+            }
+            _ => {
+                //invalidate all the entries in your way
+                //this is ok because THRESHOLD < Granuality
+                self.invaldate_following_entries(
+                    cmp::min(num_skipped_entries, NUM_ENTRIES),
+                    prev_entry,
+                );
+                self.max_update_time_gap = update_time;
+            }
+        }
+        Ok(())
+    }
+}
+
+
+#[derive(Debug, Copy, Clone)]
+#[repr(C)]
+/// Represents an Tick Tracker that has NUM_ENTRIES entries
+/// each tracking the last tick before every GRANUALITY seconds.
+pub struct TickTracker<const GRANUALITY: i64, const NUM_ENTRIES: usize, const THRESHOLD: i64> {
```

Comment:
Could you elaborate in this comment a bit more on what exactly "Tick Tracker" means? Maybe with an example?